types/nova-editor-node/index.d.ts

```diff
@@ -509,6 +575,13 @@ declare class LanguageClient {
     stop(): void;
 }
 
+type ServerOptions = {
```

Comment:
Did you mean to use this as a parameter in the LanguageClient constructor?

Reply:
Yes. Panic recommends [constructing a LanguageClient like this](https://docs.nova.app/api-reference/language-client/#constructoridentifier-name-serveroptions-clientoptions):

```javascript
var serverOptions = {
    path: path
};
var clientOptions = {
    syntaxes: ['typescript']
};
var client = new LanguageClient('typescript', 'TypeScript Language Server', serverOptions, clientOptions);

client.start();
```

Can we move `serverOptions` out of the `LanguageClient` class?:

```typescript
declare class LanguageClient {
    constructor(
        identifier: string,
        name: string,
        serverOptions: ServerOptions,
        clientOptions: { initializationOptions?: any; syntaxes: string[] },
    );

    [...]
}

interface ServerOptions {
    type?: 'stdio' | 'socket' | 'pipe';
    path: string;
    args?: string[];
    env?: { [key: string]: string };
}
```

Reply:
Yeah, this seems like an improvement to me - it gives a name to that type.